src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/SearchWriter.java

```diff
@@ -97,24 +97,22 @@ protected void addSearchFileContents(Content contentTree) {
                         .add(HtmlTree.P(contents.getContent("doclet.search.browser_info")))
                         .add(HtmlTree.SPAN(Text.of("link"))
                                 .setId(HtmlId.of("page-search-link")))
-                        .add(new HtmlTree(HtmlTag.BUTTON)
-                                .add(new HtmlTree(HtmlTag.IMG)
+                        .add(HtmlTree.BUTTON(HtmlId.of("page-search-copy"))
+                                .add(HtmlTree.of(HtmlTag.IMG)
                                         .put(HtmlAttr.SRC, pathToRoot.resolve(DocPaths.RESOURCE_FILES)
                                                                      .resolve(DocPaths.CLIPBOARD_SVG).getPath())
                                         .put(HtmlAttr.ALT, copyUrlText))
                                 .add(HtmlTree.SPAN(Text.of(copyText))
                                         .put(HtmlAttr.DATA_COPIED, copiedText))
                                 .addStyle(HtmlStyles.copy)
-                                .put(HtmlAttr.ARIA_LABEL, copyUrlText)
-                                .setId(HtmlId.of("page-search-copy")))
+                                .put(HtmlAttr.ARIA_LABEL, copyUrlText))
                         .add(HtmlTree.P(HtmlTree.INPUT(HtmlAttr.InputType.CHECKBOX, HtmlId.of("search-redirect")))
                                 .add(HtmlTree.LABEL("search-redirect",
                                         contents.getContent("doclet.search.redirect")))))
-                .add(new HtmlTree(HtmlTag.P)
+                .add(HtmlTree.of(HtmlTag.P)
                         .setId(HtmlId.of("page-search-notify"))
                         .add(contents.getContent("doclet.search.loading")))
```

Comment:
will look at that

Reply:
Yes, your suggestion matches the style of preceding usages.
I thought about adding a new overload for `HtmlTree.P` taking an `HtmlId`, but did not, because generally I think it uncommon to identify paragraphs, as compared to bigger elements like `DIV`, list, and tables.